modules/standard/Path.chpl

```diff
@@ -554,13 +569,13 @@ proc normPath(name: string): string {
     // Second case exists because we cannot go up past the top level.
     // Third case continues a chain of leading up-levels.
     if comp != parentDir || (leadingSlashes == 0 && outComps.isEmpty()) ||
-        (!outComps.isEmpty() && outComps.back() == parentDir) then
-      outComps.push_back(comp);
+        (!outComps.isEmpty() && outComps[outComps.size] == parentDir) then
+      outComps.append(comp);
     else if !outComps.isEmpty() then
-      outComps.pop_back();
+      try! outComps.pop();
   }
 
-  var result = pathSep * leadingSlashes + pathSep.join(outComps);
+  var result = pathSep * leadingSlashes + pathSep.join(outComps.toArray());
```

Comment:
It seems worth adding an overload of `join` for lists

Reply:
Or for something "iterable" which maybe one day will have a reasonable definition :)

Reply:
> Or for something "iterable"

(and could be implemented today via _iteratorRecord even though it's not user-facing... is that what you're implying?)

Reply:
Yeah, I think that today _iteratorRecord + array might cover most cases, but even there I am not so sure (because ultimately you'd want to be able to pass in a record/class that had a `proc these` but I don't think we turn that into an _iteratorRecord today....). So in practice it probably means leaving the argument totally generic and looping over it with `for`.

Reply:
> maybe one day will have a reasonable definition

this something we can expect to get with concepts/interfaces

Reply:
Aha, others thought about it before me. I conjecture that `sep.join(outComps.these())` would do it.